mapper/javabean/src/main/java/org/hibernate/search/mapper/javabean/search/loading/context/impl/JavaBeanLoadingContext.java

```diff
@@ -21,15 +21,15 @@
 
 	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
 
-	private final ProjectionHitMapper<EntityReference, Void> projectionHitMapper;
+	private final DocumentReferenceConverter<EntityReference> documentReferenceConverter;
 
-	private JavaBeanLoadingContext(ProjectionHitMapper<EntityReference, Void> projectionHitMapper) {
-		this.projectionHitMapper = projectionHitMapper;
+	private JavaBeanLoadingContext(DocumentReferenceConverter<EntityReference> documentReferenceConverter) {
+		this.documentReferenceConverter = documentReferenceConverter;
 	}
 
 	@Override
 	public ProjectionHitMapper<EntityReference, Void> getProjectionHitMapper() {
```

Comment:
Could you please also rename this method to `createProjectionHitMapper`? That would clarify that a new one is returned every time the method is called.